samples/apollo-sample/src/main/java/com/apollographql/apollo/sample/detail/GitHuntEntryDetailActivity.java

```diff
@@ -63,75 +63,89 @@ public static Intent newIntent(Context context, String repositoryFullName) {
 
   @Override protected void onCreate(@Nullable Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
-    setContentView(R.layout.activity_repository_detail);
-
-    repoFullName = getIntent().getStringExtra(ARG_REPOSITORY_FULL_NAME);
-    application = (GitHuntApplication) getApplication();
-    content = (ViewGroup) findViewById(R.id.content_holder);
-    progressBar = (ProgressBar) findViewById(R.id.loading_bar);
-    name = (TextView) findViewById(R.id.tv_repository_name);
-    description = (TextView) findViewById(R.id.tv_repository_description);
-    postedBy = (TextView) findViewById(R.id.tv_posted_by);
-
-    commentsList = (RecyclerView) findViewById(R.id.comments);
-    commentsList.setAdapter(commentsListViewAdapter);
-    commentsList.addItemDecoration(new RecyclerView.ItemDecoration() {
-      final int verticalSpaceHeight = (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, 16,
-          getResources().getDisplayMetrics());
-
-      @Override
-      public void getItemOffsets(Rect outRect, View view, RecyclerView parent, RecyclerView.State state) {
-        if (parent.getChildAdapterPosition(view) == 0) {
-          outRect.top = verticalSpaceHeight;
-          outRect.bottom = verticalSpaceHeight / 2;
-        } else if (parent.getChildAdapterPosition(view) == parent.getAdapter().getItemCount() - 1) {
-          outRect.top = verticalSpaceHeight / 2;
-          outRect.bottom = verticalSpaceHeight;
-        } else {
-          outRect.top = verticalSpaceHeight / 2;
-          outRect.bottom = verticalSpaceHeight / 2;
-        }
-      }
-    });
-    commentsList.addItemDecoration(new DividerItemDecoration(commentsList.getContext(),
-        ((LinearLayoutManager) commentsList.getLayoutManager()).getOrientation()));
-
-    getSupportActionBar().setDisplayHomeAsUpEnabled(true);
-
+    initializeUserInterface();
     fetchRepositoryDetails();
     subscribeRepoCommentAdded();
   }
 
   @Override
   public boolean onOptionsItemSelected(MenuItem item) {
-    switch (item.getItemId()) {
-      case android.R.id.home:
-        this.finish();
-        return true;
-      default:
-        return super.onOptionsItemSelected(item);
-    }
+      if (item.getItemId() == android.R.id.home) {
+          this.finish();
+          return true;
+      }
+      return super.onOptionsItemSelected(item);
```

Comment:
Maybe it's a personal preference but I tend to like `switch` better here as it conveys the idea of possibly having multiple menu items (even if there's only one right now).

Reply:
@martinbonnin absolutely, when there is multiple menu options, a `switch` is definitely the better route. However, since there isn't, and the choice is here is **binary** (either the home id or not) the Android Studio Linter displays a warning suggesting to switch to an if-statement. Whoever adds another case to the if statement can always switch to a `switch`!